flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/decorators/LabelBuilder.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.kubernetes.kubeclient.decorators;
+
+import org.apache.flink.kubernetes.utils.Constants;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Manager labels.
+ */
+public class LabelBuilder {
+
+	private Map<String, String> labels;
+
+	private LabelBuilder(){
+		this(new HashMap<>());
+	}
+
+	private LabelBuilder(Map<String, String> labels){
+		this.labels = labels;
```

Comment:
There is risk to directly take over a given map reference.
How about create a new map and put all the entries in argument `labels ` into the map?

Reply:
I will refactor the `LabelBuilder`. The hash map will be created in the construtor. All the methods will update the existing map.